File: Lenguajes_Programacion/Go/Clases.go
package main

import "fmt"

// 1. Struct Cafeteria
type Cafeteria struct {
	Nombre    string
	Ubicacion string
	Empleados int
}

func (c Cafeteria) Abrir() {
	fmt.Println("La cafetería está abierta.")
}

func (c Cafeteria) Cerrar() {
	fmt.Println("La cafetería ha cerrado.")
}

func (c Cafeteria) MostrarInfo() {
	fmt.Println("Cafetería:", c.Nombre, "- Ubicación:", c.Ubicacion, "- Empleados:", c.Empleados)
}

// 2. Struct Restaurante
type Restaurante struct {
	Nombre      string
	TipoComida  string
	Capacidad   int
}

func (r Restaurante) Abrir() {
	fmt.Println("El restaurante ha abierto sus puertas.")
}

func (r Restaurante) Cerrar() {
	fmt.Println("El restaurante ha cerrado por hoy.")
}

func (r Restaurante) MostrarDatos() {
	fmt.Println("Restaurante:", r.Nombre, "- Tipo:", r.TipoComida, "- Capacidad:", r.Capacidad)
}

// 3. Struct Libro
type Libro struct {
	Titulo string
	Autor  string
	Paginas int
}

func (l Libro) Leer() {
	fmt.Println("Leyendo el libro:", l.Titulo, "de", l.Autor)
}

func (l Libro) MarcarPagina(pagina int) {
	fmt.Println("Marcando la página", pagina, "del libro:", l.Titulo)
}

// 4. Struct Vehiculo
type Vehiculo struct {
	Marca  string
	Modelo string
	Año    int
}

func (v Vehiculo) Encender() {
	fmt.Println("El vehículo ha sido encendido.")
}

func (v Vehiculo) Apagar() {
	fmt.Println("El vehículo ha sido apagado.")
}

func (v Vehiculo) MostrarInfo() {
	fmt.Println("Vehículo:", v.Marca, v.Modelo, "- Año:", v.Año)
}

// 5. Struct Estudiante
type Estudiante struct {
	Nombre  string
	Edad    int
	Carrera string
}

func (e Estudiante) Presentarse() {
	fmt.Println("Hola, soy", e.Nombre, "y estudio", e.Carrera)
}

func (e *Estudiante) CumplirAnios() {
	e.Edad++
	fmt.Println(e.Nombre, "ahora tiene", e.Edad, "años.")
}

File: Lenguajes_Programacion/Csharp/Clases.cs
using System;

// 1. Clase Cafeteria
class Cafeteria
{
    public string Nombre;
    public string Ubicacion;
    public int Empleados;

    public void Abrir() { Console.WriteLine("La cafetería está abierta."); }
    public void Cerrar() { Console.WriteLine("La cafetería ha cerrado."); }
    public void MostrarInfo() { Console.WriteLine("Nombre: " + Nombre + ", Ubicación: " + Ubicacion + ", Empleados: " + Empleados); }
}

// 2. Clase Restaurante
class Restaurante
{
    private string nombre = "El Buen Sabor";
    private string tipoComida = "Comida típica";
    private int capacidad = 50;

    public void MostrarDatos() { Console.WriteLine("Restaurante: " + nombre + " - Tipo: " + tipoComida + " - Capacidad: " + capacidad); }
    public void Abrir() { Console.WriteLine("El restaurante ha abierto sus puertas."); }
    public void Cerrar() { Console.WriteLine("El restaurante ha cerrado por hoy."); }
}

// 3. Clase Libro
class Libro
{
    public string Titulo;
    public string Autor;
    public int Paginas;

    public void Leer() { Console.WriteLine("Leyendo el libro '" + Titulo + "' de " + Autor); }
    public void MarcarPagina(int pagina) { Console.WriteLine("Marcando la página " + pagina + " de " + Titulo); }
}

// 4. Clase Vehiculo
class Vehiculo
{
    private string marca = "Toyota";
    private string modelo = "Hilux";
    private int año = 2022;

    public void Encender() { Console.WriteLine("El vehículo ha sido encendido."); }
    public void Apagar() { Console.WriteLine("El vehículo ha sido apagado."); }
    public void MostrarInfo() { Console.WriteLine("Marca: " + marca + ", Modelo: " + modelo + ", Año: " + año); }
}

// 5. Clase Estudiante
class Estudiante
{
    private string nombre;
    private int edad;
    public string Carrera;

    public Estudiante(string nombre, int edad, string carrera)
    {
        this.nombre = nombre;
        this.edad = edad;
        this.Carrera = carrera;
    }

    public void Presentarse() { Console.WriteLine("Hola, soy " + nombre + " y estudio " + Carrera); }
    public void CumplirAños() { edad++; Console.WriteLine(nombre + " ahora tiene " + edad + " años."); }
}
